File: oim-jruby/add_user_to_group.rb
require 'java'
require 'xlclient'

include_class('java.lang.Exception') {|package,name| "J#{name}" }
include_class 'java.lang.System' 
include_class 'java.util.HashMap'
include_class('Thor.API.tcUtilityFactory') {|package,name| "OIM#{name}"}


xlclient = XLAPIClient.new
xlclient.defaultLogin
grpIntf = xlclient.getUtility('grp')

    
grpName = 'JACK_polgrp1'
usrLogin = 'JACKY1'
grpKey = xlclient.getGrpKey(grpName)
usrKey = xlclient.getUsrKey(usrLogin)

puts "usrKey = #{usrKey}"
puts "grpKey = #{grpKey}"

grpIntf.addMemberUser(grpKey, usrKey)

xlclient.close
System.exit 0

File: oim-jython/addUserToGroup.py
from xlclient import *
from java.util import *
from java.lang import *

import sys

if __name__ == '__main__':
    xlclient = XLAPIClient('xelsysadm', 'xelsysadm')
    grpIntf = xlclient.grpIntf
    
    usrLogin = 't2'
    grpName = 'res2'

    usrKey = xlclient.getUsrKey(usrLogin)
    grpKey = xlclient.getGrpKey(grpName)

    print 'usrKey = ', str(usrKey)
    print 'grpKey = ', str(grpKey)

    grpIntf.addMemberUser(grpKey, usrKey)

    xlclient.close()
    sys.exit(0)
